### PR TITLE
chore: 🤖 e2e stage in travis config

### PR DESCRIPTION
**Pull request by raduachim**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,13 +27,32 @@ jobs:
...
+        - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+        - chmod +x docker-compose
+        - sudo mv docker-compose /usr/local/bin
+        - sudo apt-get install libgconf-2-4 # cypress needs this
+        - git clone https://github.com/egendata/e2e.git ../e2e
+      install:
+        - npm ci
+        - cd ../e2e && npm ci
+      script:
+        - cd ../e2e && HOST_IP=$(ip -4 addr show docker0 | grep -Po 'inet \K[\d.]+') npm run test:local:operator
```
Comment (raduachim) [inline .travis.yml]:
```suggestion
        - cd ../e2e && npm run test:local:operator
```

File: .travis.yml [hunk 2]
```diff
@@ -23,13 +27,33 @@ jobs:
...
+
 # Flow control
 # This section defines the order in which stages are run and when to run certain stages
 stages:
   - name: lint-and-test
-    if: branch != master AND tag IS NOT present
+    if: branch != master AND tag IS NOT present AND type != pull_request
   - name: publish
     if: (branch = master AND type != pull_request)
+  - name: e2e
```
Comment (iteadam) [inline .travis.yml]: Should we not run linting and unit tests for PRs as well?

File: .travis.yml [hunk 3]
```diff
@@ -23,13 +27,33 @@ jobs:
...
+        # Install a newer version of docker-compose than the one provided by Travis
+        - sudo rm /usr/local/bin/docker-compose
+        - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+        - chmod +x docker-compose
+        - sudo mv docker-compose /usr/local/bin
+        - sudo apt-get install libgconf-2-4 # cypress needs this
+        - git clone https://github.com/egendata/e2e.git ../e2e
+      install:
+        - npm ci
+        - cd ../e2e && npm ci
```
Comment (iteadam) [inline .travis.yml]:
```suggestion
        npm ci --prefix ../e2e
```

Comment (iteadam) [reply .travis.yml]: We should not be changing directories like this. The change persists after this line

File: .travis.yml [hunk 4]
```diff
@@ -23,13 +27,33 @@ jobs:
...
+      script:
+        - cd ../e2e && npm run test:local:operator
```
Comment (iteadam) [inline .travis.yml]:
```suggestion
        - npm run --prefix ../e2e test:local:operator
```

File: .travis.yml [hunk 5]
```diff
@@ -23,13 +27,33 @@ jobs:
...
+        - npm ci --prefix ../e2e
+      script:
+        - npm run test:local:operator --prefix ../e2e
+
 # Flow control
 # This section defines the order in which stages are run and when to run certain stages
 stages:
   - name: lint-and-test
-    if: branch != master AND tag IS NOT present
+    if: branch != master AND tag IS NOT present AND type != pull_request
```
Comment (iteadam) [inline .travis.yml]:
```suggestion
    if: branch != master AND tag IS NOT present
```

File: .travis.yml (hunk 2 above)
Comment (iteadam) [reply .travis.yml]: It will always run both :+1:

**Pull request merged**